src/common.ts

```diff
@@ -100,3 +100,63 @@ export async function validateTxTip(node: GQLNodeInterface, ardriveOracle: ArDri
 
 	return tipPercentage + EPSILON >= 15 && wasValidTipRecipient;
 }
+
+export function daysDiffInEST(prev: Date, curr: Date): number {
+	const prevEstDate = dateToEST(prev);
+	const currEstDate = dateToEST(curr);
+
+	let daysCount = 0;
+	const cursorDate = new Date(prevEstDate.getTime());
+	cursorDate.setHours(0);
+	cursorDate.setMinutes(0);
+	cursorDate.setSeconds(0);
+
+	while (currEstDate.getTime() > cursorDate.getTime()) {
+		const cursorDay = cursorDate.getDate();
+		cursorDate.setDate(cursorDay + 1);
+		if (cursorDate.getTime() < currEstDate.getTime()) {
+			daysCount++;
+		}
+	}
+	return daysCount;
+}
+
+export function weeksDiffInEST(prev: Date, curr: Date): number {
+	const prevEstDate = dateToEST(prev);
+	const currEstDate = dateToEST(curr);
+
+	let weeksCount = 0;
+	const cursorDate = toSunday(prevEstDate);
+
+	while (currEstDate.getTime() > cursorDate.getTime()) {
+		const cursorDay = cursorDate.getDate();
+		const daysPerWeek = 7;
+		cursorDate.setDate(cursorDay + daysPerWeek);
+		if (cursorDate.getTime() < currEstDate.getTime()) {
+			weeksCount++;
+		}
+	}
+	return weeksCount;
+}
+
+function toSunday(date: Date) {
+	const day = date.getDay() || 7;
+	const newDate = new Date(date.getTime());
+	if (day !== 7) {
```

Comment:
You need to use 0 if you want to target sunday
```suggestion
	if (day !== 0) {
```